sklearn/linear_model/coordinate_descent.py

```diff
@@ -1218,7 +1218,9 @@ def fit(self, X, y):
         model.alpha = best_alpha
         model.l1_ratio = best_l1_ratio
         model.copy_X = copy_X
-        model.precompute = False
+        precompute = getattr(self, "precompute", None)
+        if isinstance(precompute, str) and precompute == "auto":
+            model.precompute = False
```

Comment:
Shouldn't there be an,
```py
elif isinstance(precompute, bool):
    model.precompute = precompute
```
?
Say in `Lasso` the default for dense is `precompute=False`, which means that even if `precompute=True` is passed to `LassoCV`, it would still not be used. Or am I missing something?

Reply:
I don't think it's the right fix. The _pre_fit function uses:

https://github.com/scikit-learn/scikit-learn/blob/605ae39578d7817dd75a78c746beb60b1f0f4e1a//sklearn/linear_model/base.py#L546

I would introduce a private function

```
def _get_precompute(X, precompute):
      if isinstance(precompute, str) and precompute == "auto":
           n_samples, n_features = X.shape
           return (n_samples > n_features)
      else:
           return precompute
```

that gets called in both _pre_fit and here taking as input self.precompute

Reply:
and I yes I agree we should use the same strategy for the path and the single fit at the end. Otherwise it's dangerous as seen with the issue reported.

now this being said it's quite uncommon to use a Lasso with such n_samples >> n_features @Meta95
that's why i guess the problem was never reported.

Reply:
@rth model.precompute was already set above to ``precompute`` as part of ``common_params``.
@agramfort In Lasso, we removed the "auto" function because it was not helpful according to  #3249. So I assume for a single fit setting it to ``False`` is a better choice than using the heuristic here. That would mean that if it's set to "auto", the path algorithm might use ``precompute=True`` while the last fit doesn't. But I think that makes sense, given that the path algorithm reuses results while the single fit doesn't.
That's purely based on the discussion in #3249 though.

Reply:
ok you're right. it's the right fix as auto was removed from Lasso. Now I realize that 'auto' could/should have been kept with a rule like `(n_samples > 3 * n_features)` for example (3 or more based on a bench). Anyhow +1 for MRG as it is. thx @amueller for taking a stab at this.